photoeditor/src/main/java/ja/burhanrashid52/photoeditor/GraphicHelper.java

```diff
@@ -0,0 +1,50 @@
+package ja.burhanrashid52.photoeditor;
+
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.FrameLayout;
+import android.widget.ImageView;
+
+/**
+ * Created by Burhanuddin Rashid on 18/05/21.
+ *
+ * @author <https://github.com/burhanrashid52>
+ */
+class GraphicHelper {
+    private final ViewGroup mViewGroup;
```

Comment:
My initial thought was the same. I extracted `GraphicHelper` exclusively to deal with the helper box. But then I thought of giving a generic name that might be used in the future to handle more cases related to these boxes. I understand your point in mixing two concepts In graphics.

Maybe I can move the `GraphicHelper` outside the `Graphics` and rename it as `BoxHelper`. Would it make sense?

I was having two thoughts on whether to keep `Graphic` as an abstract class or interface. That's why I needed a third-person opinion. I personally like to have a Graphic interface. But we have a repetitive view creation and listener logic that can be converted into [Template Pattern](https://www.tutorialspoint.com/design_pattern/template_pattern.htm) using abstract class.

I tried first with the interface and was not happy with the result and hence abstract class made sense to me. Just thinking out loud maybe we can use both abstract and interface Example: `Sticker extends Graphics implements IGraphic` Thoughts?

Reply:
> Maybe I can move the GraphicHelper outside the Graphics and rename it as BoxHelper. Would it make sense?

Thank you for doing this. `BoxHelper` is a better name IMO.

> I was having two thoughts on whether to keep Graphic as an abstract class or interface. That's why I needed a third-person opinion. I personally like to have a Graphic interface. But we have a repetitive view creation and listener logic that can be converted into Template Pattern using abstract class.

Yep. Speaking strictly for myself, I've found using Abstract classes and a Template Pattern can degrade if the complexity grows. Specifically, if there is a lot of needed functionality, or a very deep tree of types, the abstract classes get cluttered with tons of functionality. Due to the nature of inheritance, the abstract class functionality gets included by default in every subclass.  At some point the subclasses diverge on what they expect the superclass / abstract class functions to do, so there's proliferation. In the end it can result in a lot of tangling.

Note this isn't the case if the abstract class is an interface to a 3rd party, where the definitions are fixed and compliance with the template "contract" helps keep the user / library interface clean.

In this case, it's not complex enough to merit more work. Since this is strictly internal, we can switch the approach if it gets out of hand.

> I tried first with the interface and was not happy with the result and hence abstract class made sense to me. Just thinking out loud maybe we can use both abstract and interface Example: Sticker extends Graphics implements IGraphic Thoughts?

Since none of this is exposed to the user, I'm fine leaving it either way. If it becomes unwieldy down the line we can switch it up.

Reply:
Ah forgot to mention one thing: Generally speaking, Template Pattern / Abstract classes are harder to test, since inheritance trees cannot be replaced with mock objects.

Doesn't change my answer from above, just wanted to flag it.